projects/eval_scannet/scannet/load_scannet_data.py

```diff
@@ -0,0 +1,207 @@
+# Modified from
+# https://github.com/facebookresearch/votenet/blob/master/scannet/load_scannet_data.py
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""Load Scannet scenes with vertices and ground truth labels for semantic and
+instance segmentations."""
+import argparse
+import inspect
+import json
+import os
+
+import numpy as np
+import scannet_utils
+
+currentdir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+
+
+def read_aggregation(filename):
+    assert os.path.isfile(filename)
+    object_id_to_segs = {}
+    label_to_segs = {}
+    with open(filename) as f:
+        data = json.load(f)
+        num_objects = len(data["segGroups"])
+        for i in range(num_objects):
+            object_id = (
+                data["segGroups"][i]["objectId"] + 1
+            )  # instance ids should be 1-indexed
+            label = data["segGroups"][i]["label"]
+            segs = data["segGroups"][i]["segments"]
+            object_id_to_segs[object_id] = segs
+            if label in label_to_segs:
+                label_to_segs[label].extend(segs)
+            else:
+                label_to_segs[label] = segs
+    return object_id_to_segs, label_to_segs
+
+
+def read_segmentation(filename):
+    assert os.path.isfile(filename)
+    seg_to_verts = {}
+    with open(filename) as f:
+        data = json.load(f)
+        num_verts = len(data["segIndices"])
+        for i in range(num_verts):
+            seg_id = data["segIndices"][i]
+            if seg_id in seg_to_verts:
+                seg_to_verts[seg_id].append(i)
+            else:
+                seg_to_verts[seg_id] = [i]
+    return seg_to_verts, num_verts
+
+
+def extract_bbox(mesh_vertices, object_id_to_segs, object_id_to_label_id, instance_ids):
```

Comment:
gets per-instance bounding boxes from the mesh vertices associated with each object, from the scannet labels?

Reply:
Yes -- in practice the mesh vertices here are downsampled a bit from the full mesh (50k points/scene), following the convention of VoteNet and MMDet3d